### PR TITLE
chore: use TrustedMailServers from NodeConfig in mailserver cycle for Waku V1

### PR DESCRIPTION
**Pull request by richard-ramos**

### CODE REVIEW

File: src/app_service/tasks/marathon/mailserver/model.nim [hunk 1]
```diff
@@ -63,26 +62,34 @@ proc newMailserverModel*(vptr: ByteAddress): MailserverModel =
...
-  let fleets =
-    if defined(windows) and defined(production):
+  
+  self.wakuVersion = status_settings.getWakuVersion()
+
+  let nodeConfig = status_settings.getNodeConfig()
+  if self.wakuVersion == 2:
+    # TODO:
+    # Instead of obtaining the waku2 fleet from fleet.json, expose a method in status-go that will
+    # return the list of store nodes. (The cluster config can ontain dns-discovery urls so it cannot be
```
Comment (alaibe) [inline src/app_service/tasks/marathon/mailserver/model.nim:76]: nit ontain => contain

File: src/app_service/tasks/marathon/mailserver/model.nim [hunk 2]
```diff
@@ -63,26 +62,34 @@ proc newMailserverModel*(vptr: ByteAddress): MailserverModel =
...
+    # used to populate the list of mailservers)
+    let fleets = if defined(windows) and defined(production):
       "/../resources/fleets.json"
     else:
       "/../fleets.json"
+    let fleetConfig = readFile(joinPath(getAppDir(), fleets))
+    let fleet = newFleetModel(fleetConfig)
+    self.mailservers = toSeq(fleet.config.getMailservers(status_settings.getFleet(), true).values)
```
Comment (alaibe) [inline src/app_service/tasks/marathon/mailserver/model.nim:84]: why do we use `status_settings.getFleet()` while we already read the fleets.json file? Maybe there is a confusing naming?

Comment (alaibe) [inline src/app_service/tasks/marathon/mailserver/model.nim:84]: nit: as in python when calling function with literal I would suggest to specify the name of the args like this:
```
     self.mailservers = toSeq(fleet.config.getMailservers(status_settings.getFleet(), isWakuV2=true).values)
```
All people reading your code will praise you for making their life easer ;)

**Review by alaibe — APPROVED** [review]
Only a few nit comments! LGTM

**Review by jrainville — APPROVED** [review]

**Review by PascalPrecht — APPROVED** [review]
Changes LGTM, but would love to have @alaibe 's comment addressed as well

File: src/app_service/tasks/marathon/mailserver/model.nim (hunk 2 above)
Comment (richard-ramos) [reply src/app_service/tasks/marathon/mailserver/model.nim:84]: Indeed. It's confusing.  getFleet() returns the current fleet name stored in the settings

**Review by anastasiyaig — APPROVED** [review]

**Pull request merged**